tests/conftest.py

```
@@ -0,0 +1,102 @@
+        f.write('hi')
+
+
+@dataclass
+class LiveServer:
+    url: str
+    root: str  # 测试数据根目录
+
+
+@pytest.fixture
+def live_server():
+    """每个测试独立一份数据目录 + 独立端口的 JmServer 实例（Flask dev server 线程内运行）。"""
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)